doc/release/upcoming_changes/14882.deprecation.rst

```diff
@@ -0,0 +1,30 @@
+Using the aliases of builtin types like ``np.int`` is deprecated
+----------------------------------------------------------------
+
+For a long time, ``np.int`` has been an alias of the builtin ``int``. This is
+repeatedly a cause of confusion for newcomers, and is also simply not useful.
+
+These aliases have been deprecated. The table below shows the full list of
+deprecated aliases, along with their exact meaning. Replacing uses of items in
+the first column with the contents of the second column will work identically
+and silence the deprecation warning.
+
+In many cases, it may have been intended to use the types from the third column.
+Be aware that use of these types may result in subtle but desirable behavior#
+changes.
+
+==================  =====================  ============================
+Deprecated          Equivalent to          Possibly intended numpy type
+==================  =====================  ============================
+``numpy.bool``      ``bool``               `numpy.bool_`
+``numpy.int``       ``int``                `numpy.int_` (default int dtype), `numpy.cint` (C ``int``)
+``numpy.float``     ``float``              `numpy.float_`, `numpy.double` (equivalent)
+``numpy.complex``   ``complex``            `numpy.complex_`, `numpy.cdouble` (equivalent)
+``numpy.object``    ``object``             `numpy.object_`
+``numpy.str``       ``str``                `numpy.str_`
+``numpy.long``      ``numpy.compat.long``  `numpy.int_` (C ``long``), `numpy.longlong` (largest integer type)
+``numpy.unicode``   ``numpy.compat.str``   `numpy.unicode_`
+==================  =====================  ============================
+
+Note that for technical reasons these deprecation warnings will only be emitted
+on Python 3.7 and above.
```

Comment:
```suggestion
on Python 3.7 and above.  These warnings will be emitted only the first time when accessing
the deprecated alias.  For a large code base, it may be best to use a text search to fix all cases.
```
YMMV, I thought it might be good to point out.

Reply:
Is this even true?

Reply:
Arrg, sorry, we have to assign it for this to be true, but we never add it to the modules dict in the getter code... Unlike for testing :/...

Reply:
Yes. This is a feature, because it means that you don't have to do multiple test runs to find all the warnings, there will be one warning per occurrence.